### PR TITLE
Verifier hash

### PR DESCRIPTION
**Pull request by brentzundel**
add convenience function to verifier - create_challenge_hash
modify tests to use convenience function
fix inputs of prover::create_challenge_hash to match (slices not vectors)
add tests

### CODE REVIEW

File: libzmix/bbs/src/verifier.rs [hunk 1]
```diff
@@ -71,4 +71,35 @@ impl Verifier {
...
+    ///
+    /// # Arguments
+    /// * `proofs` - a vec of SignatureProof objects
+    /// * `proof_requests` - a corresponding vec of ProofRequest objects
+    /// * `nonce` - a SignatureNonce
+    /// * `claims` - a vec of strings the prover wishes to include in the challenge (may be empty)
+    pub fn create_challenge_hash(
+        proofs: &[SignatureProof],
+        proof_requests: &[ProofRequest],
+        claims: &[&[u8]],
```
Comment (mikelodder7) [inline libzmix/bbs/src/verifier.rs:85]: Maybe this should be optional. I'd like to use the method but I don't necessarily have any additional claims to add.

`claims: Option<&[&[u8]]>`

File: libzmix/bbs/src/prover.rs [hunk 2]
```diff
@@ -123,8 +123,8 @@ impl Prover {
     /// * `nonce` - a SignatureNonce
     /// * `claims` - a vec of strings the prover wishes to include in the challenge (may be empty)
```
Comment (kdenhartog) [inline libzmix/bbs/src/prover.rs:124]: This comment may be updated now? I think it's supposed to be an optional slice of bytes. Not recognizing if these types are the same. I'm a bit "rusty" at reading rust

image

Comment (brentzundel) [reply libzmix/bbs/src/prover.rs]: good catch, fixed in [df1086c](https://github.com/hyperledger/ursa/pull/120/commits/df1086cfaad044d82c79eb6a1f02749c3a50f2e7)

**Comment by brentzundel** [comment]
@mikelodder7 any other changes you'd like to see?

**Comment by mikelodder7** [comment]
> @mikelodder7 any other changes you'd like to see?

Nope its all good now.

**Pull request merged**